axiom/tokens_test.go

```diff
@@ -0,0 +1,256 @@
+        "datasetCapabilities": {
+            "dataset": {
+                "ingest": [
+                    "create"
+                ],
+                "query": [
+                    "read"
+                ]
+            }
+        },
+		"expiresAt": "%s",
+        "description": "test",
+        "name": "test",
+        "orgCapabilities": {
+            "apiTokens": [
+                "create"
+            ]
+        },
+		"token":"test"
+    }`, tokenTime.UTC().Truncate(time.Second).Format(time.RFC3339))
+		assert.NoError(t, err)
+	}
+	client := setup(t, "/v2/tokens/api", hf)
+
+	res, err := client.Tokens.Create(context.Background(), CreateTokenRequest{
+		Name:        "test",
+		Description: "test",
+		ExpiresAt:   tokenTime.UTC().Truncate(time.Second),
+		DatasetCapabilities: map[string]DatasetCapabilities{
+			"dataset": {
+				Ingest: []string{"create"},
+				Query:  []string{"read"},
+			},
+		},
+		OrganisationCapabilities: OrganisationCapabilities{
+			APITokens: []string{"create"},
+		},
+	})
+	require.NoError(t, err)
+
+	assert.Equal(t, exp, res)
+}
+
+func TestTokensService_Regenerate(t *testing.T) {
+	exp := &CreateTokenResponse{
+		APIToken: APIToken{
+			Name:        "test",
+			Description: "test",
+			ExpiresAt:   tokenTime.Add(24 * time.Hour).UTC().Truncate(time.Second),
+			DatasetCapabilities: map[string]DatasetCapabilities{
+				"dataset": {
+					Ingest: []string{"create"},
+					Query:  []string{"read"},
+				},
+			},
+			OrganisationCapabilities: OrganisationCapabilities{
+				APITokens: []string{"create"},
+			},
+		},
+		Token: "test",
+	}
+	hf := func(w http.ResponseWriter, r *http.Request) {
+		assert.Equal(t, http.MethodPost, r.Method)
+		assert.Equal(t, mediaTypeJSON, r.Header.Get("Content-Type"))
+
+		w.Header().Set("Content-Type", mediaTypeJSON)
+		_, err := fmt.Fprintf(w, `{
+        "datasetCapabilities": {
+            "dataset": {
+                "ingest": [
+                    "create"
+                ],
+                "query": [
+                    "read"
+                ]
+            }
+        },
+		"expiresAt": "%s",
+        "description": "test",
+        "name": "test",
+        "orgCapabilities": {
+            "apiTokens": [
+                "create"
+            ]
+        },
+		"token":"test"
+    }`, tokenTime.Add(24*time.Hour).UTC().Truncate(time.Second).Format(time.RFC3339))
+		assert.NoError(t, err)
+	}
+	client := setup(t, "/v2/tokens/api/test/regenerate", hf)
+
+	res, err := client.Tokens.Regenerate(context.Background(), "test", RegenerateTokenRequest{
+		ExistingTokenExpiresAt: tokenTime,
+		NewTokenExpiresAt:      tokenTime.Add(24 * time.Hour),
+	})
+	require.NoError(t, err)
+
+	assert.Equal(t, exp, res)
+}
+
+func TestTokensService_Delete(t *testing.T) {
+	hf := func(w http.ResponseWriter, r *http.Request) {
+		assert.Equal(t, http.MethodDelete, r.Method)
+
+		w.WriteHeader(http.StatusNoContent)
+	}
+
+	client := setup(t, "/v2/tokens/api/testID", hf)
```

Comment:
Mismatches the API reference given as a comment on the services implementation.

Reply:
Addressed on comment above